addons/account/models/account_move.py

```diff
@@ -2799,51 +2810,78 @@ class AccountMoveLine(models.Model):
         default=1.0, digits='Product Unit of Measure',
         help="The optional quantity expressed by this line, eg: number of product sold. "
              "The quantity is not a legal requirement but is very useful for some reports.")
-    price_unit = fields.Float(string='Unit Price', digits='Product Price')
+    price_unit = fields.Float(
+        string='Unit Price',
+        store=True, readonly=False,
+        compute='_compute_price_unit',
+        digits='Product Price')
     discount = fields.Float(string='Discount (%)', digits='Discount', default=0.0)
     debit = fields.Monetary(string='Debit', default=0.0, currency_field='company_currency_id')
     credit = fields.Monetary(string='Credit', default=0.0, currency_field='company_currency_id')
-    balance = fields.Monetary(string='Balance', store=True,
+    balance = fields.Monetary(
+        string='Balance',
+        store=True,
         currency_field='company_currency_id',
         compute='_compute_balance',
         help="Technical field holding the debit - credit in order to open meaningful graph views from reports")
-    cumulated_balance = fields.Monetary(string='Cumulated Balance', store=False,
-        currency_field='company_currency_id',
-        compute='_compute_cumulated_balance',
-        help="Cumulated balance depending on the domain and the order chosen in the view.")
-    amount_currency = fields.Monetary(string='Amount in Currency', store=True, copy=True,
+    amount_currency = fields.Monetary(
+        string='Amount in Currency',
         help="The amount expressed in an optional other currency if it is a multi-currency entry.")
-    price_subtotal = fields.Monetary(string='Subtotal', store=True, readonly=True,
+    price_subtotal = fields.Monetary(
+        string='Subtotal',
+        store=True,
+        compute='_compute_price_subtotal',
         currency_field='currency_id')
-    price_total = fields.Monetary(string='Total', store=True, readonly=True,
+    price_total = fields.Monetary(
+        string='Total',
+        store=True,
+        compute='_compute_price_subtotal',
         currency_field='currency_id')
     reconciled = fields.Boolean(compute='_compute_amount_residual', store=True)
     blocked = fields.Boolean(string='No Follow-up', default=False,
         help="You can check this box to mark this journal item as a litigation with the associated partner")
     date_maturity = fields.Date(string='Due Date', index=True, tracking=True,
         help="This field is used for payable and receivable journal entries. You can put the limit date for the payment of this line.")
-    currency_id = fields.Many2one('res.currency', string='Currency', required=True)
-    partner_id = fields.Many2one('res.partner', string='Partner', ondelete='restrict')
-    product_uom_id = fields.Many2one('uom.uom', string='Unit of Measure', domain="[('category_id', '=', product_uom_category_id)]")
+    currency_id = fields.Many2one(
```

Comment:
NOTE: `currency_id` still has a very bad behavior. Indeed this field is required by doesn't have any default value because this value could only be given by a computed method (dependency to move's company currency). However, making this field required has 2 side effects:
- remove `required=True` on the field and check its value using a python constraints.
- `_sql_constraints` should also be replaced by a python constraints.
=> To discuss with qdp